infra/README.md

```diff
@@ -0,0 +1,75 @@
+  accounts/             Root module for IaC and IAM resources
+  [app_name]/           Application directory: infrastructure for the main application
+  modules/              Reusable child modules
+  networks/             Account level network config (shared across all apps, environments, and terraform workspaces)
+```
+
+Each application directory contains the following:
+
+```text
+  app-config/         Application-level configuration for the application resources (different config for different environments)
+  build-repository/   Docker image repository for the application (shared across environments and terraform workspaces)
+  database/           Configuration for database (different config for different environments)
+  service/            Configuration for containers (load balancer, application service) (different config for different environments)
+```
+
+### 🏗️ Project architecture
+
+#### 🥞 Infrastructure layers
+
+The infrastructure template is designed to operate on different layers.
+
+- Account layer
+- Network layer
+- Build repository layer (per application)
+- Database layer (per application)
+- Service layer (per application)
+
+### 🔀 Project workflow
+
+This project relies on Make targets in the [root Makefile](../Makefile), which in turn call shell scripts in [./bin](../bin). The shell scripts call terraform commands. Many of the shell scripts are also called by the [Github Actions CI/CD](../.github/workflows).
+
+Backend configuration is saved as [`.tfbackend`](https://developer.hashicorp.com/terraform/language/settings/backends/configuration#file) files. Most `.tfbackend` files are named after the environment. For example, the `[app_name]/service` infrastructure resources for the `dev` environment are configured via `dev.s3.tfbackend`. Resources for a module that are shared across environments, such as the build-repository, use `shared.s3.tfbackend`. Resources that are shared across the entire account (e.g. /infra/accounts) use `<account name>.<account id>.s3.tfbackend`.
+
+Generally you should use the Make targets or the underlying bin scripts, but if you want to directly use the `terraform` cli, you need to pass in the `.tfbackend` file to init commands and the `.tfvars` file to other commands:
+
+```sh
+infra/portal/service$ terraform init -backend-config=dev.s3.tfbackend
+infra/portal/service$ terraform apply -var="environment_name=dev"
+```
+
+## 💻 Development
+
+### 1️⃣ First time initialization
+
+To set up this project for the first time (aka it has never been deployed to the target AWS account):
+
+1. Follow the steps as outlined in the [infrastructure documentation README](../docs/infra/README.md) for:
+    1. Configure the project
+    1. Set up infrastructure developer tools
+    1. Set up AWS account
+1. [Set up the network](../docs/infra/set-up-network.md)
+1. Go through the [infrastructure documentation README](../docs/infra/README.md) application steps for each [app]
```

Comment:
optional nit: I might be in the minority here, but i like markdown numbered lists to reflect the actual numbers, like 1,2,3,... not 1, 1, 1, ... The reason is because the spirit of markdown is that it should be readable as plaintext in addition to being readable by readers that can interpret markdown with rich formatting. the arguments i can see for keeping it as 1, 1, 1, ... are (1) it's easier to edit, and (2) it makes PR changes easier to review, but for (1), [vscode plugins](https://github.com/yzhang-gh/vscode-markdown) can automatically update all the numbers for you just like in rich editors like google docs, and for (2) i think github's diffs have gotten good about highlighting just the part of the line that changes, so it's easy to see when just the numbers change across many lines rather than the whole lines.

anyways don't feel strongly about this at all but just my 2 cents